### PR TITLE
create tenant API

### PR DESCRIPTION
**Pull request by bboozzoo**
@mendersoftware/rndity

### CODE REVIEW

**Comment by bboozzoo** [comment]
Should be complete now.

File: model/tenant.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package model
+
+type NewTenant struct {
+	ID string
```
Comment (mchalski) [inline model/tenant.go]: tenant_id?

File: store/mongo/tenant_datastore.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func NewTenantStoreMongo(db *DataStoreMongo) *TenantStoreMongo {
+	return &TenantStoreMongo{
+		db: db,
+	}
+}
+
+func (ts *TenantStoreMongo) MigrateTenant(ctx context.Context, id string) error {
+	return ts.db.MigrateTenant(ctx, DbVersion, id)
```
Comment (mchalski) [inline store/mongo/tenant_datastore.go]: why not just expose DbVersion to the outside world and use directly in App?

File: model/tenant.go (hunk 1 above)
Comment (bboozzoo) [reply model/tenant.go]: That would be a repetition of `tenant` in the name of the field fields of a structure that is already named after 'tenant'.

File: store/mongo/tenant_datastore.go (hunk 2 above)
Comment (bboozzoo) [reply store/mongo/tenant_datastore.go]: It is unfortunately and it's defied in `store/mongo` I believe. It's also used in the `main` package and in datastore tests where `store/mongo` is imported. `App` on the other hand is in `useradm` package which does not import `store/mongo` and only works with dependencies that were passed from main.Importing `store/mongo` there would feel a bit off.

**Pull request merged**